target/linux/ath79/image/generic-tp-link.mk

```diff
@@ -99,6 +99,16 @@ define Device/tplink_tl-wr1043nd-v1
 endef
 TARGET_DEVICES += tplink_tl-wr1043nd-v1
 
+define Device/tplink_tl-wr842n-v1
+  $(Device/tplink-8m)
+  ATH_SOC := ar7241
+  DEVICE_TITLE := TP-LINK TL-WR842N/ND v1
+  DEVICE_PACKAGES := kmod-usb-core kmod-usb2 kmod-usb-ledtrig-usbport
+  TPLINK_HWID := 0x8420001
+  SUPPORTED_DEVICES += tl-wr842n-v1
```

Comment:
It's set to `SUPPORTED_DEVICES := tplink,tl-wr842n-v1` by default in [target/linux/ath79/image/Makefile](https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/image/Makefile#L63)

> AR71xx target used "tl-mr3420" as board id so force flag is needed if upgrading from old target.

What about adding `SUPPORTED_DEVICES += tl-mr3420` ? Would that work when upgrading from `ar71xx` to `ath79`? Can you please test it?

Reply:
> Having said that, there are other targets with redundant SUPPORTED_DEVICES assignment so makefiles need cleaning.

Yes, it's very likely. I wasn't aware about this nice feature myself, I've learned about it from another PR review. Anyway, feel free to send a cleanup patch(es).

> upgraded WNDR3700v2 which also needed -F

If you've device such device and spot this kind of problem, it would be nice if you could help fix it. Patches for this kind of issues are more then welcome, since it would be nice to have seamless migration ar71xx to ath79 experience.

> Such mistake would probably lead to destroying ART partition etc.

Ok, that's quite scary, so better don't do that. BTW, nice mess :-)